agent/connect_ca_endpoint.go

```diff
@@ -63,9 +63,9 @@ func (s *HTTPServer) ConnectCAConfigurationSet(resp http.ResponseWriter, req *ht
 	s.parseDC(req, &args.Datacenter)
 	s.parseToken(req, &args.Token)
 	if err := decodeBody(req.Body, &args.Config); err != nil {
-		resp.WriteHeader(http.StatusBadRequest)
-		fmt.Fprintf(resp, "Request decode failed: %v", err)
-		return nil, nil
```

Comment:
This looks like a fun unrelated bug that you fixed.

Reply:
Yep, because of this pattern the test asserting no error returned was not doing anything and took me a while to figure out why a later assertion was failing...